Source/common/SNTConfigurator.m

```diff
@@ -549,6 +560,21 @@ - (void)setRemountUSBMode:(NSArray<NSString *> *)args {
   return args;
 }
 
+- (NSDictionary<NSString *, SNTRule *> *)staticRules {
+  NSArray *currentRules = self.configState[kStaticRules];
+  if (currentRules.hash != self.cachedStaticRulesHash) {
```

Comment:
How expensive is this hashing? For a really large set I can see the cost adding up since we need to hash for every exec. Instead of initializing lazily here we should move the parsing closer to when the rules are added to the profile. We could add a new property for the unprocessed rules and KVO observe changes. We could do the parsing there. This method turns into a fast getter for the current set of rules.